### DISCUSSION TITLE
MiniDexed serialmididevice error

### DISCUSSION THREAD

**Discussion by CerebralKaos** [probonopd/MiniDexed#745] [Q&A] [answered]
I have just build a MiniDexed based on a Raspberry Pi 3B+ and have incorporated the encoder and a i2C DAC.  When I try launch MiniDexed I get the following error: 

serialmididevice: Serial.Read() error: -1.

I'm not using a serial MIDI connection.  I am connecting via MIDI over USB with an Arturia Controller and I enabled MIDI Dump which has indicated the Pi is receiving MIDI information when connected up to a HDMI Monitor.

Any idea what is causing the issue please?
![IMG_6017](https://github.com/user-attachments/assets/4fc55c2d-0ffc-4d4f-a2c3-34785aad6d41)

**Comment by probonopd** [comment]
These messages are really only intended for developers.
Normal users should keep the debugging off, as it worsens the sound.
Does this impact the operation of MiniDexed in any way?
Maybe the message can be ignored?

**Comment by diyelectromusic** [comment]
There is often a spurious serial spurt if serial MIDI is enabled (which it always is) but not used, so that can be ignored.  We ought to make the reporting more useful really, but as it doesn't affect anything and in most cases only happens once on power up, its not really considered a major issue.

One day I'll take a proper look at why it occurs :)

Kevin

**Comment by CerebralKaos** [comment]
Hi Kevin,It doesn’t boot into Dexed. It just stops there and doesn’t boot into MiniDexed.  The LCD screen is lit but blank.RegardsShaneOn 10 Nov 2024, at 3:19 AM, Kevin ***@***.***> wrote:﻿
There is often a spurious serial spurt if serial MIDI is enabled (which it always is) but not used, so that can be ignored.  We ought to make the reporting more useful really, but as it doesn't affect anything and in most cases only happens once on power up, its not really considered a major issue.
One day I'll take a proper look at why it occurs :)
Kevin

**Comment by CerebralKaos** [comment]
I have worked out that the problem is the LCD I am using is not interfaced to the Pi properly, so I have ordered a1602 with i2c with the potentiometer on the back.  MiniDexed is loading the voices and I was able to get some limited sound out of the headphone amp.

**Reply by diyelectromusic** [reply]
Ah - well done for spotting it :)  Glad you got something going.

Kevin

**Reply by CerebralKaos** [reply]
Attached the new 16X2 i2C LCD, still no display on it and am positive I have connected correctly using the four pin connections.  It lights up but no text is displayed. Do I have to change the code in the minidexed.ini or somewhere else?

**Reply by CerebralKaos** [reply, marked as answer]
Solved.

Just had to add:
If you are using a LCDC1602 with an i2c "backpack" board, then you need to set LCDI2CAddress=0x27 (or another address your i2c "backpack" board is set to) in minidexed.ini
